File: java-console/brasil-burger-manager/src/main/java/com/brasilburger/domain/entities/Complement.java
package com.brasilburger.domain.entities;

import com.brasilburger.domain.entities.enums.CategorieArticle;
import com.brasilburger.domain.entities.enums.TypeComplement;

/**
 * Entite Complement
 */
public class Complement extends Article {
    private TypeComplement type;
    private Integer prix;  // Prix obligatoire pour un complement

    /**
     * Constructeur par defaut
     */
    public Complement() {
        super();
        this.setCategorie(CategorieArticle.COMPLEMENT);
    }

    /**
     * Constructeur avec parametres
     */
    public Complement(String code, String libelle, String imagePublicId, TypeComplement type, Integer prix) {
        super(code, libelle, imagePublicId);
        this.setCategorie(CategorieArticle.COMPLEMENT);
        this.type = type;
        this.setPrix(prix);
    }

    // ===================================
    // Implementation de la methode abstraite
    // ===================================

    @Override
    public Integer getPrix() {
        return prix;
    }

    // ===================================
    // Methodes metier
    // ===================================

    /**
     * Valide que le prix est positif
     */
    private void validerPrix(Integer prix) {
        if (prix == null || prix <= 0) {
            throw new IllegalArgumentException("Un complement doit avoir un prix positif");
        }
    }

    /**
     * Verifie si c'est une boisson
     */
    public boolean estBoisson() {
        return type == TypeComplement.BOISSON;
    }

    /**
     * Verifie si ce sont des frites
     */
    public boolean estFrites() {
        return type == TypeComplement.FRITES;
    }

    // ===================================
    // Getters et Setters
    // ===================================

    public TypeComplement getType() {
        return type;
    }

    public void setType(TypeComplement type) {
        this.type = type;
    }

    public void setPrix(Integer prix) {
        validerPrix(prix);
        this.prix = prix;
    }
}

File: csharp-client/BrasilBurger.Client/src/BrasilBurger.Client.Domain/Entities/Complement.cs
using BrasilBurger.Client.Domain.Common;
using BrasilBurger.Client.Domain.Enums;

namespace BrasilBurger.Client.Domain.Entities;

public class Complement : Article
{
    private Complement() { } // EF

    public Complement(string code, string libelle, string imagePublicId, TypeComplement typeComplement, int? prix)
        : base(code, libelle, imagePublicId)
    {
        TypeComplement = typeComplement;
        Prix = prix;
    }

    public override CategorieArticle Categorie => CategorieArticle.COMPLEMENT;

    public override int? GetPrix() => Prix;

    public void ChangerType(TypeComplement typeComplement) => TypeComplement = typeComplement;

    public void FixerPrix(int? prix)
    {
        if (prix is not null) Guard.Positive(prix.Value, nameof(prix));
        Prix = prix;
    }
}
